### ISSUE TITLE
Expose Sigstore verification configuration in PolicyServers

### ISSUE THREAD

**Issue by viccuad** [kubewarden/kubewarden-controller#137]
With the work done on policy-server, kwctl, ClusterAdmissionPolicies can be verified with Sigstore keys and annotations. Expose those to users.

## Acceptance criteria

PolicyServer CRD has 2 new spec fields:
- `spec.enableVerification`, `bool`: Optional, default false if not present. Enables Sigstore verification. If `spec.verificationSecret` is present, use that information for verification. If not, perform Keyless verification with default settings. Once enabled, all policies associated with the PolicyServer must be signed with the given keys and annotations. 
- `spec.verificationSecret`, `string`: Optional, default empty. Name of the Kubernetes Secret on the same ns as the PolicyServer, with the contents of  [verification.yml](https://github.com/kubewarden/policy-server/wiki/Configuration#verification---verification-path) file, containing verification keys and annotations.

**Comment by viccuad** [comment]
Another option is to have those 2 configuration fields on the controller, and have them affect all PolicyServers managed by the controller.

**Comment by flavio** [comment]
I think it's better to have these fields inside of the `PolicyServer` CRD. By doing that we allow the operators to have multiple PolicyServer instances with different settings. For example, one PolicyServer with verification turned on, another with verification disabled and yet another with verification enabled but using a different set of keys.

I just have some concerns about the immutable fields. I know why you're doing that, but I wonder if this is going more of an hindrance to the end user. Morover, as a user, I can find ways to circumvent this immutability. I can have the controller load the keys from an already existing Secret and then I can keep changing the contents of this Secret.

The immutability of these fields forces me, as an user, to remove the PolicyServer and then recreate it with the right set of keys/annotations I want. This would cause all the ClusterAdmissionPolicy associated with this PolicyServer to be removed and would force me (the operator) to re-create them.

I think it's better to leave these fields mutable, the controller will then redeploy everything with the new setting.

**Comment by ereslibre** [comment]
I agree with @flavio on the location of this fields. I think they only belong to the PolicyServer CRD. The reason is beacause while the setting is clear and explicit on the CRD itself, it's not so clear the behavior/settings of the controller acting on those resources. So I favour not tying the controller to this settings at all. In my opinion, the PolicyServer should contain all the information that describes how the policy server behaves.

I also think that both fields can be optional (as the default). If the user wants to force `enableVerification` to be `true` at all times, we can provide a policy for that, along with `verificationSecret`. It could be part of the bootstrapping process of Kubewarden, so from that point on, everything is ensured to be what we expect as users. This policies can even ensure the contents of the secrets, e.g. by making sure that at least N keys are present in the secret, so even if a bad actor has rights to modify the secret, the request will be rejected by the policy, because those minimum N keys were not present on the new secret.

In short: I think is better to not make the controller add any own logic to the reconciliation, and just behave based on what it sees on the PolicyServer CRD, and I think we can leave fields mutable and provide a couple (or a single) policies to control the immutability of these fields, also with extra value in terms of the secret containing the keys.

**Comment by viccuad** [comment]
Thanks both on the comments, settings on PolicyServer CRD then :).
Edited the acceptance criteria to remove the immutable mentions.

**Comment by viccuad** [comment]
> I can have the controller load the keys from an already existing Secret and then I can keep changing the contents of this Secret.

I'm still going forward with mutable fields for enabling verification and secret name, but just realized that [k8s 1.21+ allows for immutable secrets](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable).
